src/main/scala-3/com/raquo/airstream/split/MatchValueObservable.scala

```diff
@@ -0,0 +1,52 @@
+package com.raquo.airstream.split
+
+import com.raquo.airstream.core.{Observable, BaseObservable}
+import scala.annotation.compileTimeOnly
+
+/** `MatchSingletonObservable` served as macro's data holder for macro expansion.
+ *
+ * For example:
+ *
+ * ```scala
+ * fooSignal.splitMatch
+ *   .splitValue(Tar)(tarSignal => renderTarNode(tarSignal))
+ * ```
+ * 
+ * will be expanded sematically into:
+ *
+ * ```scala
+ * MatchTypeObservable.build[*, *, *, Baz](
+ *   fooSignal,
+ *   Nil,
+ *   handlerMap,
+ *   ({ case Tar => Tar })
+ * )
+ * ```
+ *
+ * and then into:
+ *
+ * ```scala
+ * MatchSplitObservable.build(
+ *   fooSignal,
+ *   ({ case Tar => Tar }) :: Nil,
+ *   handlerMap
+ * )
+ * ```
+ */
+
+opaque type MatchValueObservable[Self[+_] <: Observable[_], I, O, V0, V1] = Unit
```

Comment:
`V0` and `V1` should never be different, but we cannot just do `MatchValueObservable[Self[+_] <: Observable[_], I, O, V]` because after type erasure, compiler cannot differentiates `MatchValueObservable[Self[+_] <: Observable[_], I, O, V]` from `MatchTypeObservable[Self[+_] <: Observable[_], I, O, T]`.

Reply:
TBH the implications of this are over my head, but it sounds like it's just an unfortunate implementation detail that we have to deal with. In that case, fine by me, _I_ certainly don't have better ideas on this.

I think we're all good now, thanks for all the updates!

I am super pumped for these new operators, this will be one of the first things that I will merge for 18.x, proooobably in a few weeks (but apologies in advance if my 18.x gets delayed a bit, honestly, it's possible, but I'll do my best to carve out some time for it asap).

Reply:
This ambiguity is due to the usage of opaque types. I changed them back to value classes to remove this implementation hack. It should be easier to read and understand now.